docs/source/devguides/pycharm/Pycharm-Dev-Environment.rst

```diff
@@ -0,0 +1,118 @@
+.. _Pycharm-Dev-Environment:
+
+Pycharm Development Environment
+===============================
+
+Pycharm is an IDE dedicated to developing python projects. It provides coding
+assistance and easy access to debugging tools as well as integration with
+py.test. It is a popular tool for working with VOLTTRON.
+Jetbrains provides a free community version that can be downloaded from
+https://www.jetbrains.com/pycharm/
+
+
+Open Pycharm and Load VOLTTRON
+------------------------------
+
+When launching Pycharm for the first time we have to tell it where to find the
+VOLTTRON source code. If you have already cloned the repo then point Pycharm to
+the cloned project. Pycharm also has options to access remote repositories.
+
+Subsequent instances of Pycharm will automatically load the VOLTTRON project.
+
+|Open Pycharm|
+|Load Volttron|
+
+
+Set the Project Interpreter
+---------------------------
+
+This step should be completed after running the bootstrap script in the VOLTTRON
+source directory. Pycharm needs to know which python environment it should  use
+when running and debugging code. This also tells Pycharm where to find python
+dependencies.
+
+|Set Project Interpreter|
+
+
+Running the VOLTTRON Process
+----------------------------
+
+If you are not interested in running the VOLTTRON process itself in Pycharm then
+this step can be skipped.
+
+In **Run > Edit Configurations** create a configuration that has
+`<your source dir>/env/bin/volttron` in the script field, `-vv` in the script
+parameters field (to turn on verbose logging), and set the working directory to
+the top level source directory.
+
+VOLTTRON can then be run from the Run menu.
+
+|Run Settings|
+
+
+Running an Agent
+----------------
+
+Before running any agents in Pycharm we must allow VOLTTRON to talk to them.
```

Comment:
This does not need to be done any longer.  See pycharm-launch.py for what is necessary